src/vision/status_quo/barbara_writes_a_runtime_agnostic_lib.md

```diff
@@ -0,0 +1,125 @@
+making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories
+[cannot be wrong], only inaccurate). Alternatively, you may wish to
+[add your own status quo story][htvsq]!
+
+## The story
+
+Barbara and Alan work at AmoolgeSoft, where many teams are switching from Java to Rust. These teams
+have many different use cases and various adoption stories. Some teams are happy users of tokio,
+others happy users of async-std, and others still are using custom runtimes for highly specialized
+use cases.
+
+Barbara is tasked with writing a library for a custom protocol, SLOW (only in use at AmoogleSoft)
+and enlists the help of Alan in doing so. Alan is already aware that [not all libraries in Rust work
+with all runtimes][nalirwwar]. Alan and Barbara start by writing a parser which works on
+`std::io::Read` and get their tests working with `String`s. After this they contemplate the question
+of how to accept a TCP connection.
+
+### Incompatible `AsyncRead` traits
+
+Alan asks Barbara what is the async equivalent is of `std::io::Read`, and Barbara sighs and says
+that there isn't one. Barbara brings up tokio's and the [futures crate]'s versions of `AsyncRead`.
+Barbara decides not to talk about `AsyncBufRead` for now.
+
+Barbara and Alan decide to use the future's `AsyncRead` for no other reason other than it is
+runtime-agnostic. Barbara tells Alan not to worry as they can translate between the two. With some
+effort (**TODO:** maybe link a story about `Pin`? which one?) they convert their parser to using
+`AsyncRead`.
+
+Alan, excited about the progress they've made, starts working on hooking this up to actual TCP
+streams. Alan looks at async-std and tokio and notices their interfaces for TCP are quite different.
+Alan waits for Barbara to save the day.
+
+Barbara helps abstract over TCP listener and TCP stream (**TODO:** code example). One big hurdle is
+that tokio uses `AsyncRead` from their own crate and not the one from `futures` crate.
+
+### Task spawning
+
+After getting the TCP handling part working, they now want to spawn tasks for handling each incoming
+TCP connection. Again, to their disappointment, they find that there's no runtime-agnostic way to do
+that.
+
+Unsure on how to do this, they do some searching and find the [`agnostik`] crate. They reject it
+because this only supports N number of runtimes and their custom runtime is not one of them.
+However it gives them the idea to provide a trait for specifying how to spawn tasks on the runtime.
+Barbara points out that this has disadvantage of working against orphan rules meaning that either
+they have to implement the trait for all known runtimes (defeating the purpose of the exercise) or
+force the user to use new types.
+
+They punt on this question by implementing the trait for each of the known runtimes. They're
+disappointed that this means their library actually isn't runtime agnostic.
+
+### The need for timers
```

Comment:
I'd like to mention that they could add a timer method to their runtime trait and much like spawning require every runtime to provide an implementation, but they decide not to do this to try to keep this trait as simple as possible.

Reply:
That's already there, isn't it? Or do you mean I should change the wording?